wpilibcExamples/src/main/cpp/examples/ArmBot/cpp/RobotContainer.cpp

```diff
@@ -60,6 +60,6 @@ void RobotContainer::DisablePIDSubsystems() {
   m_arm.Disable();
 }
 
-frc2::Command* RobotContainer::GetAutonomousCommand() {
-  return nullptr;
+frc2::CommandPtr RobotContainer::GetAutonomousCommand() {
+  return frc2::CommandPtr(nullptr);
```

Comment:
Something doesn't smell right here.  If CommandPtr(nullptr) is a standard thing to work with, why are the `std::optional<>`'s used in other places.  Should this also return a std::optional?

Reply:
That's a good point. I made it optional because the command template had it as such. CommandPtr has a bool operator to check if the unique_ptr is valid so I will use this operator instead. I think this makes more sense than an optional anyways. Should this change be propagated to the command template too?

Reply:
I think so.  It’s confusing to have two layers where there might be an absence.

Reply:
`CommandPtr(nullptr)` shouldn't be valid... Is there a case where it's proven to compile?

Reply:
I set the m_autonomousCommand in robot.h to be cmd::None() by default however in functions returning no auto, CommandPtr(nullptr) is returned.

Reply:
`CommandPtr(nullptr)` should _not_ compile. If it does, that's an unintentional bug and should be fixed.

As for `Robot.m_autonomousCommand`, the initial value should be "no command" (=empty optional), not "command doing nothing" (=`cmd::None()`).

Reply:
Why shouldn't it? unique_ptr can be set to nullptr and is there not a use case where a CommandPtr is set to null until a Command is moved into it? Either way, I do agree that GetAutonomousCommand() should probably return a cmd::None() rather than a CommandPtr(nullptr) so I will update the functions to do so. It is still probably a good idea to keep the check for the validity of the CommandPtr before running it in Robot.cpp though.

Reply:
This is a fair point, there should be consideration for absence of command, not a command that does nothing. An optional does this. I am currently working on the other requests, but will implement optional shortly.

Reply:
Should GetAutonomousCommand return an optional and be std::nullopt if no command or should it return cmd::None()?

Reply:
`cmd::None()`.
Theoretically it might be good for all examples to have at least some minimal auto, but that might be out of scope for this.

Reply:
I'll leave it as cmd::None() for this pr but I may open another soon to add maybe a minimal auto for examples that can have it.

Reply:
Leaving it as it is now with `nullptr` as a `Command*` also works.